lib/concurrent/set.rb

```diff
@@ -0,0 +1,44 @@
+require 'concurrent/utility/engine'
+require 'concurrent/thread_safe/util'
+require 'set'
+
+module Concurrent
+  if Concurrent.on_cruby?
+
+    # Because MRI never runs code in parallel, the existing
+    # non-thread-safe structures should usually work fine.
+
+    # @!macro [attach] concurrent_Set
+    #
+    #   A thread-safe subclass of Set. This version locks against the object
+    #   itself for every method call, ensuring only one thread can be reading
+    #   or writing at a time. This includes iteration methods like `#each`.
+    #
+    #   @note `a += b` is **not** a **thread-safe** operation on
+    #   `Concurrent::Set`. It reads Set `a`, then it creates new `Concurrent::Set`
+    #   which is concatenation of `a` and `b`, then it writes the concatenation to `a`.
+    #   The read and write are independent operations they do not form a single atomic
+    #   operation therefore when two `+=` operations are executed concurrently updates
```

Comment:
Unfinished sentence: `updates` ... `may be lost` from https://github.com/ruby-concurrency/concurrent-ruby/blob/df482db36caf1b0c1d69a8ff97a2407469e1e315/lib/concurrent/array.rb#L21